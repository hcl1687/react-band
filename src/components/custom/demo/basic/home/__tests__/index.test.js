import { mount, render } from 'enzyme'
import React from 'react'
import homeFactory from '../index.entry'

const Home = homeFactory()

describe('demo/basic/home', () => {
  it('should render correctly', () => {
    const __ = key => key
    const his = {
      push: key => key
    }
    const wrapper = render(<Home __={__} history={his} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should invoke i18n translator', () => {
    const __ = jest.fn()
    const his = {
      push: jest.fn()
    }
    const wrapper = mount(
      <Home __={__} history={his} />
    )
    expect(wrapper.find('div').length).toBe(2)
    expect(wrapper.find('button').length).toBe(1)
    expect(__).toHaveBeenCalled()
    wrapper.find('button').simulate('click')
    expect(his.push).toHaveBeenCalled()
  })
})
