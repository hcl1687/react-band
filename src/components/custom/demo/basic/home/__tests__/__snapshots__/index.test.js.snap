// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`demo/basic/home should render correctly 1`] = `
<div>
  <div>
    home
  </div>
  <button>
    toTest
  </button>
</div>
`;
